hikari/impl/entity_factory.py

```diff
@@ -728,10 +728,15 @@ def deserialize_channel(
         guild_id: undefined.UndefinedOr[snowflakes.Snowflake] = undefined.UNDEFINED,
     ) -> channel_models.PartialChannel:
         channel_type = payload["type"]
-        if channel_model := self._guild_channel_type_mapping.get(channel_type):
-            return channel_model(payload, guild_id=guild_id)
+        if guild_channel_model := self._guild_channel_type_mapping.get(channel_type):
+            return guild_channel_model(payload, guild_id=guild_id)
 
-        return self._dm_channel_type_mapping[channel_type](payload)
+        if dm_channel_model := self._dm_channel_type_mapping.get(channel_type):
+            return dm_channel_model(payload)
+
+        return channel_models.UnrecognisedChannel(
```

Comment:
Would it be easier to just fall back to a partial channel and log that it's an unexpected channel type was received?

Reply:
This is exactly what this is doing. Forgot the logging part tho, so will add